src/server/worker/datum/datum.go

```diff
@@ -64,7 +60,39 @@ func CreateSets(dit Iterator, storageRoot string, setSpec *SetSpec, upload func(
 	}); err != nil {
 		return err
 	}
-	return createSet(metas, storageRoot, upload)
+	if len(metas) > 0 {
+		return createSet(metas, storageRoot, upload)
+	}
+	return nil
+}
+
+func shouldCreateSetFunc(setSpec *SetSpec) func(*Meta) bool {
+	switch {
+	case setSpec.Number > 0:
+		var num int64
+		return func(meta *Meta) bool {
+			num++
+			if num >= setSpec.Number {
+				num = 0
+				return true
+			}
+			return false
+		}
+	case setSpec.SizeBytes > 0:
+		var size int64
+		return func(meta *Meta) bool {
+			for _, input := range meta.Inputs {
+				size += int64(input.FileInfo.SizeBytes)
+			}
+			if size >= setSpec.SizeBytes {
+				size = 0
+				return true
+			}
+			return false
+		}
+	default:
+		panic("must set a set spec field")
```

Comment:
Do we really want to panic here? It seems like that will just put the worker in a crashloop that won't give any feedback to the user about why the pipelines isn't proceeding.

Reply:
This shouldn't ever happen because we default to evenly distributing the datums in the registry. I'm going to change this to a default of 1 when it is not set just in case. That default will apply at the datum package level, and the default of distributing the datums evenly will still apply at the transform package level.